File: Habitica/src/main/java/com/habitrpg/android/habitica/models/user/SuppressedModals.java
package com.habitrpg.android.habitica.models.user;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class SuppressedModals extends RealmObject {

    @PrimaryKey
    private String userId;

    Preferences preferences;
    private Boolean streak, raisePet, hatchPet, levelUp;

    public Boolean getStreak() {
        return streak;
    }

    public void setStreak(Boolean streak) {
        this.streak = streak;
    }

    public Boolean getRaisePet() {
        return raisePet;
    }

    public void setRaisePet(Boolean raisePet) {
        this.raisePet = raisePet;
    }

    public Boolean getHatchPet() {
        return hatchPet;
    }

    public void setHatchPet(Boolean hatchPet) {
        this.hatchPet = hatchPet;
    }

    public Boolean getLevelUp() {
        return levelUp;
    }

    public void setLevelUp(Boolean levelUp) {
        this.levelUp = levelUp;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }
}

File: Habitica/src/main/java/com/habitrpg/android/habitica/models/user/SuppressedModals.kt
package com.habitrpg.android.habitica.models.user

import com.habitrpg.android.habitica.models.BaseObject
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass

@RealmClass(embedded = true)
open class SuppressedModals : RealmObject(), BaseObject {
    var streak: Boolean? = null
    var raisePet: Boolean? = null
    var hatchPet: Boolean? = null
    var levelUp: Boolean? = null
}